src/api/uploads/BaseMultiput.js

```diff
@@ -0,0 +1,80 @@
+/**
+ * @flow
+ * @file Multiput upload base class
+ * @author Box
+ */
+import Base from '../Base';
+import type { MultiputConfig, Options } from '../../flowTypes';
+
+const DEFAULT_MULTIPUT_CONFIG: MultiputConfig = {
+    digestReadahead: 5, // How many parts past those currently uploading to precompute digest for
+    initialRetryDelayMs: 5000, // Base for exponential backoff on retries
+    maxRetryDelayMs: 60000, // Upper bound for time between retries
+    parallelism: 5, // Maximum number of parts to upload at a time
+    requestTimeoutMs: 120000, // Idle timeout on part upload, overall request timeout on other requests
+    // eslint-disable-next-line max-len
+    retries: 5 // How many times to retry requests such as upload part or commit. Note that total number of attempts will be retries + 1 in worst case where all attempts fail.
+};
+
+class BaseMultiput extends Base {
+    config: MultiputConfig;
+    sessionEndpoints: Object;
+    canConsoleLog: boolean;
+
+    /**
+     * [constructor]
+     *
+     * @param {Options} options
+     * @param {Object} sessionEndpoints
+     * @param {MultiputConfig} [config]
+     * @return {void}
+     */
+    constructor(options: Options, sessionEndpoints: Object, config?: MultiputConfig): void {
+        super(options);
+
+        this.config = config || DEFAULT_MULTIPUT_CONFIG;
+        this.sessionEndpoints = sessionEndpoints;
+        this.canConsoleLog = !!options.consoleLog && !!window.console && !!window.console.log;
```

Comment:
Is this still needed? Since this.consoleLog will be a noop, you can just call it without check of existance.

Reply:
Yes, because if you can't console log, then we won't call the function to generate the console message which could potentially be costly. see `consoleLogFunc`